### ISSUE TITLE
`@test_broken` doesn't work within `@testset` on 0.5

### ISSUE THREAD

**Issue by omus** [JuliaLang/julia#20368]
Using the [`@test_broken`](http://docs.julialang.org/en/stable/stdlib/test/#broken-tests) macro within an `@testset` will cause an exception to be raised when the expression evaluates to false in Julia 0.5:

Julia 0.5
```julia
julia> using Base.Test

julia> @testset begin
           @test_broken 1 == 2
       end
Test Summary: | Broken  Total
  test set    |      1      1
ERROR: Some tests did not pass: 0 passed, 0 failed, 0 errored, 1 broken.
 in finish(::Base.Test.DefaultTestSet) at ./test.jl:495
 in macro expansion; at ./test.jl:679 [inlined]
 in anonymous at ./<missing>:?
```

Julia 0.6 has the correct behaviour:

```julia
julia> using Base.Test

julia> @testset begin
           @test_broken 1 == 2
       end
Test Summary: | Broken  Total
test set      |      1      1
Base.Test.DefaultTestSet("test set",Any[Test Broken
Expression: 1 == 2
],0,false)
```

**Comment by omus** [comment]
cc: @StefanKarpinski

**Comment by KristofferC** [comment]
Fixed on 0.6. Seems unlikely to be backported.

**Comment by tkelman** [comment]
fixed in 0.5.2 by https://github.com/JuliaLang/julia/pull/21087
